tests/support/helpers.py

```diff
@@ -1067,50 +1067,135 @@ def decorator(*args, **kwargs):
     return decorator
 
 
+@requires_system_grains
+def runs_on(grains=None, **kwargs):
+    '''
+    Skip the test if grains don't match the values passed into **kwargs
+    if a kwarg value is a list then skip if the grains don't match any item in the list
+    '''
+    def decorator(caller):
+        @functools.wraps(caller)
+        def wrapper(cls):
+            for kw, value in kwargs.items():
+                if isinstance(value, list):
+                    if not any(str(grains.get(kw)).lower() != str(v).lower() for v in value):
+                        cls.skipTest('This test does not run on {}={}'.format(kw, grains.get(kw)))
+                else:
+                    if str(grains.get(kw)).lower() != str(value).lower():
+                        cls.skipTest('This test runs on {}={}, not {}'.format(kw, value, grains.get(kw)))
+            return caller(cls)
+        return wrapper
+    return decorator
+
+
+@requires_system_grains
+def not_runs_on(grains=None, **kwargs):
+    '''
+    Reverse of `runs_on`.
+    Skip the test if any grains match the values passed into **kwargs
+    if a kwarg value is a list then skip if the grains match any item in the list
+    '''
+    def decorator(caller):
+        @functools.wraps(caller)
+        def wrapper(cls):
+            for kw, value in kwargs.items():
+                if isinstance(value, list):
+                    if any(str(grains.get(kw)).lower() == str(v).lower() for v in value):
+                        cls.skipTest('This test does not run on {}={}'.format(kw, grains.get(kw)))
+                else:
+                    if str(grains.get(kw)).lower() == str(value).lower():
+                        cls.skipTest('This test does not run on {}={}, got {}'.format(kw, value, grains.get(kw)))
+            return caller(cls)
+        return wrapper
+    return decorator
+
+
+def check_required_sminion_attributes(sminion_attr, *required_items):
```

Comment:
I understand why Daniel wanted this as a top level function, specially now that its used on 2 different helpers, but let's prefix it with a `_`.
This way people aren't tempted to use it directly and will create the right `requires_salt_*` helper function.

Reply:
I noticed this function might be usable in conftest.py since it contains an identical code block there, but since helpers.py imports from unittest is that a good idea?  Maybe there is another place this can go so that it is usable by conftest.py

Reply:
For now, leave it here.
Once we go PyTest, we'll rearrange.